woocommerce/woocommerce-paypal-payments
# Move the subscription product in cart existence and the vaulting enabled check into the callbacks.

**Issue**: #624

---

### Description

When Vaulting is disabled, the Smart Buttons will temporarily display in the Mini Cart anyway after adding the subscription product to the cart.
Updating the cart quantity or adding another product resolves the behavior.

ALSO
- When there is a regular product in mini-cart together with a subscription product.
- Then removing the subscription product should make the button appear for the regular product left in the mini-cart.

The PR will solve these problems by moving the subscription product in cart existence and the vaulting enabled check into  the button renderer hook callback so that the mini-cart ajax refresh will take it into the consideration.

### Steps to Test




1. Activate the Storefront theme.
2. Disable the Vaulting.
3. Add subscription product to cart from the shop page (not the single product page).
4. Hover over mini cart → smart buttons will appear.
5. Adding another product to the cart or updating the quantity via the Cart page removes the smart buttons from the Mini Cart

**ALSO**
6. When there is a regular product in mini-cart together with a subscription product.
7. Then removing the subscription product should make the button appear for the regular product left in the mini-cart.

---

Closes #624 .
